lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,398 @@
+  Color get keyColor => new HexColor(data['key_color']);
+
+  /// When was this [Multireddit] created.
+  DateTime get createdUtc => new DateTime.fromMillisecondsSinceEpoch(
+      data['created_utc'].round() * 1000,
+      isUtc: true);
+
+  /// The [IconName] associated with this multireddit.
+  ///
+  /// Refer to [IconName]'s Enum definition for more information.
+  /// If this information is not provided, will return null.
+  IconName get iconName => IconName.values.firstWhere(
+      (e) =>
+          e.toString() ==
+          ('IconName.' + _convertToCamelCase(data['icon_name'])),
+      orElse: () => null);
+
+  List<SubredditRef> get subreddits {
+    final subredditList = [];
+    subredditList.addAll(data['subreddits']
+        .map((subreddit) => new SubredditRef.name(reddit, subreddit['name'])));
+    return subredditList;
+  }
+
+  /// The [Redditor] associated with this multireddit.
+  RedditorRef get author => _author;
+  RedditorRef _author;
+
+  /// The displayName given to the [Multireddit].
+  String get displayName => data['display_name'];
+
+  /// The infoPath will be used to uniquely identify this multireddit.
+  String get infoPath => _infoPath ?? '/';
+  String _infoPath;
+
+  /// The visibility of this multireddit.
+  ///
+  /// Refer to [Visibility]'s Enum definition for more information.
+  /// If this information is not provided, will return null.
+  Visibility get visibility => Visibility.values.firstWhere(
+      (e) => e.toString() == ('Visibility.' + data['visibility']),
+      orElse: () => null);
+
+  /// The weightingScheme of this multireddit.
+  ///
+  /// Refer to [WeightingScheme]'s Enum definition for more information.
+  /// If this information is not provided, will return null.
+  WeightingScheme get weightingScheme => WeightingScheme.values.firstWhere(
+      (e) => e.toString() == ('WeightingScheme.' + data['weighting_scheme']),
+      orElse: () => null);
+
+  /// Does the currently authenticated [User] have the privilege to edit this
+  /// multireddit.
+  bool get canEdit => data['can_edit'];
+
+  /// Does this multireddit require visitors to be over the age of 18.
+  bool get over18 => data['over_18'];
+
   Multireddit.parse(Reddit reddit, Map data) : super(reddit) {
+    if (!data['data'].containsKey('name')) {
+      throw new DRAWUnimplementedError();
+    }
     setData(this, data['data']);
+    _author = new RedditorRef.name(
+        reddit, data['data']['path']?.split('/')[_redditorNameInPathIndex]);
+    _infoPath = _generateInfoPath(data['data']['name'], _author.displayName);
+  }
+
+  // Returns valid info_path for multireddit with name `name`.
+  static String _generateInfoPath(String name, String user) =>
+      apiPath['multireddit']
+          .replaceAll(_multiredditRegExp, name)
+          .replaceAll(_userRegExp, user);
+
+  // Returns `str` in camel case.
+  static String _convertToCamelCase(String str) {
+    if (str == null) {
```

Comment:
Should we ever hit this? If not, throw.

Reply:
So we actually do need this, because there are cases where the JSON response for some of the enums is completely empty. In those cases, this is the desired behaviour.